Try based lookup methods

manigandham opened the pull request:
Refers to #31 

Added new `TryCity` and `TryCountry` methods that don't throw an exception when the IP address cannot be found. This avoids using exceptions for flow control.

manigandham commented:
I can add Try... methods to the other lookup types as well if wanted.

oschwald commented:
Thanks! Please do add them for all the database types along with corresponding unit tests.

manigandham commented:
Updated.

oschwald commented on GeoIP2/DatabaseReader.cs:
```diff
@@ -147,6 +195,18 @@ public AnonymousIPResponse AnonymousIP(string ipAddress)
+        /// <param name="ipAddress">The IP address.</param>
+        /// <param name="response">The <see cref="AnonymousIPResponse" />.</param>
+        /// <returns>A <see cref="bool" /> describing whether the IP address was found.</returns>
+        public bool TryAnonymousIP(string ipAddress, out AnonymousIPResponse response)
```
For some of these, there is only a method that takes a `string` IP and for others the method takes only an `IPAddress`. We should have both or at least be consistent.

manigandham commented:
All set now. You're right, I missed some of the methods in that class.

Added the missing methods, organized the class better to have the base execute methods at the end and refactored the database reader unit tests too.

oschwald commented:
Thanks! I merged this and did a beta release.
